backend/src/graphql/resolver/AdminResolver.test.ts

```diff
@@ -0,0 +1,1054 @@
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [new GraphQLError('No Creation found!')],
+                  }),
+                )
+              })
+            })
+
+            describe('amount of creation is too high', () => {
+              it('throws an error', async () => {
+                variables.creationDate = new Date().toString()
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [
+                      new GraphQLError(
+                        'The amount (2000 GDD) to be created exceeds the available amount (1000 GDD) for this month.',
+                      ),
+                    ],
+                  }),
+                )
+              })
+            })
+
+            describe('creation is valid', () => {
+              it('returns an array of the open creations for the last three months', async () => {
+                variables.amount = new Decimal(200)
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    data: {
+                      createPendingCreation: [1000, 1000, 800],
+                    },
+                  }),
+                )
+              })
+            })
+
+            describe('second creation surpasses the available amount ', () => {
+              it('returns an array of the open creations for the last three months', async () => {
+                variables.amount = new Decimal(1000)
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [
+                      new GraphQLError(
+                        'The amount (1000 GDD) to be created exceeds the available amount (800 GDD) for this month.',
+                      ),
+                    ],
+                  }),
+                )
+              })
+            })
+          })
+        })
+
+        describe('createPendingCreations', () => {
+          // at this point we have this data in DB:
+          // bibi@bloxberg.de: [1000, 1000, 800]
+          // peter@lustig.de: [1000, 600, 1000]
+          // stephen@hawking.uk: [1000, 1000, 1000] - deleted
+          // garrick@ollivander.com: [1000, 1000, 1000] - not activated
+
+          const massCreationVariables = [
+            'bibi@bloxberg.de',
+            'peter@lustig.de',
+            'stephen@hawking.uk',
+            'garrick@ollivander.com',
+            'bob@baumeister.de',
+          ].map((email) => {
+            return {
+              email,
+              amount: new Decimal(500),
```

Comment:
THis whole construct is bound to fail, what if there is two pending creations for the same amount for the same month for the same user?